src/js/tools/metroconfig.ts

```diff
@@ -158,26 +158,83 @@ function withSentryDebugId(config: MetroConfig): MetroConfig {
   };
 }
 
+type ResolverThreeParams = (
+  context: CustomResolutionContext,
+  moduleName: string,
+  platform: string | null,
+) => Resolution;
+type ResolverFourParams = (
+  context: CustomResolutionContext,
+  moduleName: string,
+  platform: string | null,
+  realModuleName?: string,
+) => Resolution;
+
 /**
  * Includes `@sentry/replay` packages based on the `includeWebReplay` flag and current bundle `platform`.
  */
 export function withSentryResolver(config: MetroConfig, includeWebReplay: boolean | undefined): MetroConfig {
-  const originalResolver = config.resolver?.resolveRequest;
+  const originalResolver = config.resolver?.resolveRequest as ResolverThreeParams | ResolverFourParams;
+
+  const hasSentryReplay = (platform: string | null, moduleName: string): boolean => {
+    return (
+      (includeWebReplay === false || (includeWebReplay === undefined && platform !== 'web')) &&
+      moduleName.includes('@sentry/replay')
+    );
+  };
+
+  let resolver: ResolverThreeParams | ResolverFourParams;
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
```

Comment:
Can we avoid this by checking `context.resolveRequest === sentryResolverRequest` inside of the `sentryResolverRequest`?

Than we could have one sentryResolverRequest implementation.

Reply:
we now have one SentryResolverRequest implementation

Reply:
> Can we avoid this by checking context.resolveRequest === sentryResolverRequest inside of the sentryResolverRequest?

I ment can we avoid `require(metro)` to check the version? Requiring the metro package doesn't have to be accurate due to how hoisting of node modules works. 

We had issues with this in the past when requiring `defaultSerializer` from metro.

If we instead check `context.resolveRequest === sentryResolverRequest` we know that we are running in the old metro version. This should be safer as we are not relying on the hoister module to be the correct one.